src/utils/warmup.rs

```diff
@@ -0,0 +1,139 @@
+//! Provides functions for initializing and "warming up" the database, such as setting snap categories.
+#![allow(dead_code, unused_imports)]
+
+use snapd::api::convenience::SnapNameFromId;
+use snapd::api::find::FindSnapByName;
+use snapd::{SnapdClient, SnapdClientError};
+use sqlx::{Acquire, Executor, Row};
+use thiserror::Error;
+use tracing::error;
+
+use crate::app::AppContext;
+use crate::features::pb::chart::Category;
+use crate::utils::{Infrastructure, Migrator};
+
+/// Errors that can happen while warming up the database
+#[derive(Error, Debug)]
+pub enum WarmupError {
+    /// Errors from `snapd-rs`
+    #[error("an error occurred when calling snapd: {0}")]
+    SnapdError(#[from] SnapdClientError),
+    /// An error that occurred in category updating
+    #[error("an error occurred with the DB when getting categories: {0}")]
+    CategoryDBError(#[from] sqlx::Error),
+    /// An unknown error occurred
+    #[error("an unknown error occurred")]
+    Unknown,
+}
+
+/// Convenience function for getting categories by their snap ID, since it takes multiple API calls
+pub(crate) async fn snapd_categories_by_snap_id(
+    client: &SnapdClient,
+    snap_id: &str,
+) -> Result<Vec<Category>, WarmupError> {
+    let snap_name = SnapNameFromId::get_name(snap_id.into(), client).await?;
```

Comment:
I realize this is on the snapd client side, but I feel it would be worth it to hide some of the internal implementation from the rating center consuming the client (ie: that we have to aggregate multiple snapd calls to get the name by id via convenience functions) and that the rust snapd client api should just expose a `SnapNameFromId` function that just takes an Id and handles the rest?

Reply:
I agree, there's history about why this decision was made that amounts to "I had to stop working on the snapd client early", but I wouldn't mind revisiting it to clean it up.